projects/igniteui-angular/src/lib/calendar/calendar.component.html

```diff
@@ -4,49 +4,99 @@
+            #monthsBtn
+            (keydown)="onActiveViewYearKB(getViewDate(obj.index), $event, obj.index)"
+            (click)="onActiveViewYear(getViewDate(obj.index), obj.index, $event)"
+            class="igx-calendar-picker__date">
+            {{ formattedMonth(getViewDate(obj.index)) }}
+        </span>
+
+        <span
+            tabindex="0"
+            role="button"
+            [attr.aria-label]="(getViewDate(obj.index) | date: 'yyyy') + ' ' + resourceStrings.igx_calendar_select_year"
+            #yearsBtn
+            (keydown)="onActiveViewDecadeKB($event, getViewDate(obj.index), obj.index)"
+            (click)="onActiveViewDecade(getViewDate(obj.index), obj.index)"
+            class="igx-calendar-picker__date">
+            {{ formattedYear(getViewDate(obj.index)) }}
+        </span>
 </ng-template>
 
-<div *ngIf="selection === 'single' && hasHeader" class="igx-calendar__header">
-    <h5 class="igx-calendar__header-year">{{ formattedYear(headerDate) }}</h5>
-    <h2 class="igx-calendar__header-date">
+<header
+    aria-labelledby="igx-aria-calendar-title-month igx-aria-calendar-title-year"
+    class="igx-calendar__header"
+    *ngIf="selection === 'single' && hasHeader">
+
+    <h5 id="igx-aria-calendar-title-year" class="igx-calendar__header-year">
+        {{ formattedYear(headerDate) }}
+    </h5>
+
+    <h2 id="igx-aria-calendar-title-month" class="igx-calendar__header-date">
         <ng-container *ngTemplateOutlet="headerTemplate ? headerTemplate : defaultHeader; context: headerContext">
         </ng-container>
     </h2>
-</div>
+</header>
 
-<div *ngIf="isDefaultView" class="igx-calendar__body" [@animateView]="activeView" (@animateView.done)="viewRendered($event)" (swiperight)="previousMonth()"
+<div *ngIf="isDefaultView"  class="igx-calendar__body" [@animateView]="activeView" (@animateView.done)="viewRendered($event)" (swiperight)="previousMonth()"
     (swipeleft)="nextMonth()" (pointerdown)="suppressBlur()">
-    <div class="igx-calendar-picker">
-        <div tabindex="0" class="igx-calendar-picker__prev" #prevMonthBtn
-            igxCalendarScrollMonth [startScroll]="startPrevMonthScroll" [stopScroll]="stopMonthScroll"  [ngStyle]="{
-                    'min-width.%': 100/(monthsViewNumber*7)
-                }">
-            <igx-icon fontSet="material">keyboard_arrow_left</igx-icon>
+    <section class="igx-calendar-picker">
+        <span tabindex="0" class="igx-calendar__aria-off-screen">
+            <ng-container *ngIf="selection === 'multi'">
+                {{ resourceStrings.multi_selection }}
+            </ng-container>
+            <ng-container *ngIf="selection === 'range'">
+                {{ resourceStrings.range_selection }}
+            </ng-container>
+            calendar
+            {{ monthsViewNumber && monthsViewNumber > 1 ?  (resourceStrings.with + ' ' + monthsViewNumber + ' ' + resourceStrings.date_pickers) : ''}}
```

Comment:
Same thing as with `of`  just merge those into a single string like ` with {0} date pickers`